packages/web/src/components/createCell.tsx

```diff
@@ -111,103 +212,115 @@ const isDataEmptyArray = (data: DataObject) => {
-          } else {
-            console.warn(
-              `If you're using Apollo Client, check for its debug logs here in the console, which may help explain the error.`
-            )
-            throw new Error(
-              'Cannot render Cell: reached an unexpected state where the query succeeded but `data` is `null`. If this happened in Storybook, your query could be missing fields; otherwise this is most likely a GraphQL caching bug. Note that adding an `id` field to all the fields on your query may fix the issue.'
-            )
-          }
-        }}
-      </Query>
+  /**
+   * A Cell declaratively renders components based on the lifecycle of a GraphQL query.
+   *
+   * If the return's confusing, conider that a Cell's usually used in a Page like so:
+   *
+   * ```js
+   * import PostCell from 'src/components/PostCell'
+   *
+   * const PostsPage = ({ id }) => {
+   *   return <PostCell id={id} />
+   * }
+   * ```
+   *
+   * `PostCell` is actually this component.
+   */
+  function NamedCell(props: React.PropsWithChildren<CellProps>) {
+    /**
+     * Right now, Cells don't render `children`.
+     */
+    const { children: _, ...variables } = props
+
+    const options = beforeQuery(variables)
+
+    const { error, loading, data, ...queryRest } = useQuery(
+      typeof QUERY === 'function' ? QUERY(options) : QUERY,
+      options
     )
+
+    const commonProps = {
+      updating: loading,
+      ...queryRest,
+      ...props,
+    }
+
+    if (error) {
+      if (Failure) {
+        return (
+          <Failure
+            error={error}
+            errorCode={error.graphQLErrors?.[0]?.extensions?.['code'] as string}
+            {...commonProps}
+          />
+        )
+      } else {
+        throw error
+      }
+    } else if (data) {
+      const afterQueryData = afterQuery(data)
+
+      if (isEmpty(data, { isDataEmpty }) && Empty) {
+        return <Empty {...{ ...afterQueryData, ...commonProps }} />
+      } else {
+        return <Success {...{ ...afterQueryData, ...commonProps }} />
+      }
+    } else if (loading) {
+      return <Loading {...{ ...queryRest, ...props }} />
+    } else {
+      /**
+       * There really shouldn't be an `else` here, but like any piece of software,
+       * GraphQL clients have bugs.
+       *
+       * If there's no `error` and there's no `data` and we're not `loading`,
+       * something's wrong. Most likely with the cache.
+       */
+      console.warn(
+        `If you're using Apollo Client, check for its debug logs here in the console, which may help explain the error.`
+      )
+      throw new Error(
+        'Cannot render Cell: reached an unexpected state where the query succeeded but `data` is `null`. If this happened in Storybook, your query could be missing fields; otherwise this is most likely a GraphQL caching bug. Note that adding an `id` field to all the fields on your query may fix the issue.'
+      )
+    }
```

Comment:
May be a discussion for another time, but is there any reason not to just render `<Failure />` here as well?

Reply:
Hmmm... you're right. Failure would make sense!